map/bookmark_manager.cpp

```diff
@@ -1865,20 +1866,16 @@ void BookmarkManager::LoadMetadata()
     return;
 
   Metadata metadata;
-  std::vector<uint8_t> jsonStr;
-
-  auto const charPtr = [&jsonStr]()
-  {
-    return reinterpret_cast<char const *>(jsonStr.data());
-  };
+  std::string json;
 
   try
   {
-    jsonStr = base::ReadFile(metadataFilePath);
-    if (jsonStr.empty())
+    FileReader(metadataFilePath).ReadAsString(json);
+
+    if (json.empty())
       return;
 
-    coding::DeserializerJson des(charPtr());
+    coding::DeserializerJson des(json);
```

Comment:
So, if the file contains an invalid JSON, we also have a potential memory overrun here.
Otherwise, IDK how DeserializerJson knows when to stop reading :)

Reply:
It parses from a null-terminated string:

```
  explicit DeserializerJson(std::string const & source)
    : m_jsonObject(source), m_json(m_jsonObject.get())

/////////////
  explicit Json(std::string const & s) { ParseFrom(s); }
  void ParseFrom(std::string const & s) { ParseFrom(s.c_str()); }
  void ParseFrom(char const * s)
  {
    json_error_t jsonError;
    m_handle.AttachNew(json_loads(s, 0, &jsonError));
    if (!m_handle)
      MYTHROW(Exception, (jsonError.line, jsonError.text));
  }

```

Reply:
Yes, or it stops _naturally_ when JSON is ok (like it worked before).